app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormatExpr.scala

```diff
@@ -105,6 +131,16 @@ object TextConstraint {
   val defaultWholeDigits = 11
   val defaultFactionalDigits = 2
 
+  object JsonExtensions {
+    def withDefault[A](key: String, default: A)(implicit writes: Writes[A]) =
+      __.json.update((__ \ key).json.copyFrom((__ \ key).json.pick orElse Reads.pure(Json.toJson(default))))
+  }
+  implicit val numberRead: OFormat[Number] = new OFormat[Number] {
```

Comment:
Is this `OFormat[Number]` needed? It compiles fine without it and its implementation throws exception, so it seems not too useful.